### ISSUE TITLE
can not use get_job()

### ISSUE THREAD

**Issue by sky-sea** [pycontribs/jenkinsapi#471]
```python
from jenkinsapi.jenkins import Jenkins

if __name__ == '__main__':
    server = Jenkins('http://172.29.88.114:8989/jenkins', username='xxx', password='xxx')
    #server.build_job('deploy-EN-cloud-mail',{'VERSION':'1.0.0', 'DEPLOY_PATH':'/home/cloud/jenkins2/tomcat/webapps/jenkins/deploy_script'})
    #print server.get_running_builds()
    #print server.get_build_console_output('deploy-EN-cloud-mail', 19)
    server.get_job('test')


Traceback (most recent call last):
  File "F:\eclipsespace\test\demo.py", line 9, in <module>
    server.get_job('deploy-EN-cloud-mail')
  File "C:\Python27\lib\site-packages\jenkinsapi\jenkins.py", line 139, in get_job
    return self.jobs[jobname]
  File "C:\Python27\lib\site-packages\jenkinsapi\jobs.py", line 81, in __getitem__
    return Job(None, job_name, self.jenkins)
  File "C:\Python27\lib\site-packages\jenkinsapi\job.py", line 78, in __init__
    JenkinsBase.__init__(self, self.url)
  File "C:\Python27\lib\site-packages\jenkinsapi\jenkinsbase.py", line 35, in __init__
    self.poll()
  File "C:\Python27\lib\site-packages\jenkinsapi\job.py", line 134, in poll
    data = super(Job, self).poll(tree=tree)
  File "C:\Python27\lib\site-packages\jenkinsapi\jenkinsbase.py", line 59, in poll
    data = self._poll(tree=tree)
  File "C:\Python27\lib\site-packages\jenkinsapi\jenkinsbase.py", line 69, in _poll
    return self.get_data(url, tree=tree)
  File "C:\Python27\lib\site-packages\jenkinsapi\jenkinsbase.py", line 78, in get_data
    response = requester.get_url(url, params)
  File "C:\Python27\lib\site-packages\jenkinsapi\utils\requester.py", line 100, in get_url
    return requests.get(self._update_url_scheme(url), **requestKwargs)
  File "C:\Python27\lib\site-packages\requests\api.py", line 71, in get
    return request('get', url, params=params, **kwargs)
  File "C:\Python27\lib\site-packages\requests\api.py", line 57, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\Python27\lib\site-packages\requests\sessions.py", line 460, in request
    prep = self.prepare_request(req)
  File "C:\Python27\lib\site-packages\requests\sessions.py", line 393, in prepare_request
    hooks=merge_hooks(request.hooks, self.hooks),
  File "C:\Python27\lib\site-packages\requests\models.py", line 294, in prepare
    self.prepare_url(url, params)
  File "C:\Python27\lib\site-packages\requests\models.py", line 357, in prepare_url
    raise InvalidURL("Invalid URL %r: No host supplied" % url)
requests.exceptions.InvalidURL: Invalid URL u'http:///jenkins/test/api/python': No host supplied
```

**Comment by ScofieldXu** [comment]
look at jenkinsapi/job.py method _find_job_url, I think header 'Location' doesn't contain baseurl, so we just get an invalid url. I just try to modify a few codes and not get an exception.
```
utl_split = urlparse.urlsplit(self.jenkins.baseurl)
utl = urlparse.urlunsplit(
    [
        url_split.schema,
        url_split.netloc,
        search_result.headers['Location'],
        url_split.query,
        url_split.fragment
    ]
}
return url
```

**Comment by lechat** [comment]
That method of finding job url is fairly recent change. Originally `jenkins.poll()` was returning job url and I've changed that to minimize amount of data Jenkins object fetches. I am hesitant of returning that old code back, because new code is much more performant for cases when Jenkins has thousands of jobs.

Another alternative would be trying to get job by url like `http://jenkins/jobs/<job_name>`, but this will not work with folders...

It is sad, that Jenkins API doesn't have a sure way to find if job is in Jenkins or not.

**Comment by mhfsmachado** [comment]
I just changed the _find_job_url on job.py to verify if the headers['Location'] of jenkins is not including the base url and add it if not.
Different versions of jenkins have different behaviours on this so I think  this is the best way to prevent this

code:
        _......
        result_url = search_result.headers['Location']
        if self.jenkins.baseurl not in result_url:
            result_url= '{base}/{search_url}'.format(
            base=self.jenkins.baseurl,
            search_url=search_result.headers['Location']
        )
        return result_url_

**Comment by lechat** [comment]
Opened issue with Jenkins: https://issues.jenkins-ci.org/browse/JENKINS-37576

**Comment by lechat** [comment]
Restored old way of working with jobs. Will do PYPI release later today.

**Comment by lechat** [comment]
Released v0.30.1 with the fix.

**Comment by writetovaseem** [comment]
@lechat  - which version of https://pypi.python.org/pypi/jenkinsapi release has this fix?
 Looks like fix is not avaiable in latest - jenkinsapi (0.3.6)
